ext/liquid_c/tokenizer.c

```diff
@@ -127,6 +128,10 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
             cursor++;
         }
     }
+    if(token->type == TOKEN_RAW && tokenizer->trim_whitespace) {
+        token->trim_whitespace = 1;
+    }
+    tokenizer->trim_whitespace = ((token->type == TOKEN_VARIABLE || token->type == TOKEN_TAG) && token->str[token->length - 3] == '-');
```

Comment:
I'm actually refactoring this entire pull request around this as your right its much better to extend the tokenizer to look forward 3 characters rather than its current 2 so it can see the white space control character and remove any need for the code to ever look backwards.